[slider] removed react-draggable dependency (fixes IE!)

igorbt opened the pull request:
Fixing this bug https://github.com/callemall/material-ui/issues/708, I ended up that it's better to entirely remove react-draggable dependency. With this change I saw also it seems to be more responsive.
Also, this fixes desynchronization between the mouse position and slider handle.

A (very) little breaking change in the onDragStop and onDragStart APIs - it no longer provides second argument - ui. I think it was useless anyway because it had handle offset in pixels. I cannot think about a scenario when someone really need this. I you don't agree, I can try to add it back by calculating it.

igorbt committed:
removed react-draggable dependency

igorbt committed:
removed react-draggable dependency

igorbt commented:
Fixes https://github.com/callemall/material-ui/issues/1745 - see https://github.com/callemall/material-ui/issues/1745#issuecomment-146448088.

igorbt committed:
fixed wrong value in onChange when setState is async

igorbt committed:
fixed wrong value in onChange when setState is async

igorbt commented:
@yongxu, @shaurya947, @KapJI can you please take a look here. I saw you are the last contributors to the Slider component, so maybe you can make a quick review. 
The thing is that we use Slider in our product and we had a lot of issues with it (and material-ui issue tracker also if full of slider bugs) and this PR fixed all of them for us.

KapJI commented:
I reviewed this PR and tested it locally. Now handle and cursor don't diverge. Seems good to me. Good job, @igorbt!

KapJI commented on src/slider.jsx:
```diff
@@ -428,9 +476,9 @@ const Slider = React.createClass({
-    let value = this._alignValue(this._percentToValue(percent));
-    if (this.props.onChange) this.props.onChange(e, value);
+    this.setPercent(percent, () => {
+      if (this.props.onChange) this.props.onChange(e, this.state.value);
```
This will call onChange many times with same value during dragging. I think it will be better to call onChange only when value changes.

igorbt replied:
Thanks @KapJI for review and for this finding. You are right and I just fixed it, but from setPercent, it seemed more convenient there for me.

igorbt commented:
@yongxu, thanks for review!
You are right, percent is not always the same as this.state.percent. This is why in setState passed value is normalized before putting in state:

    setPercent(percent, callback) {
        let value = this._alignValue(this._percentToValue(percent));
        let { min, max } = this.props;
        let alignedPercent = (value - min) / (max - min);
        this.setState({value: value, percent: alignedPercent}, callback);
    },

KapJI commented:
@yongxu comparing percents difference with step is not a good idea, because step is measured in absolute value, not in percents.

igorbt committed:
updating value in state only if it really changes

so avoiding unnecessary re-rendering and unnecessary calling of onChange from _updateWithChangeEvent.

igorbt committed:
updating value in state only if it really changes

so avoiding unnecessary re-rendering and unnecessary calling of onChange from _updateWithChangeEvent.

yongxu commented:
Thanks @KapJI , you were right I didn't think about it. it should be something like```(step/(max-min))```
@igorbt have solved this in another way, I think it should be good enough!

igorbt committed:
Removed unneded percent checking

igorbt committed:
Removed unneded percent checking

igorbt commented:
@yongxu you are right, just removed that line.

igorbt commented:
@shaurya947 can you please also have a look and merge this PR?

shaurya947 committed:
Merge pull request #1884 from callemall/react-0.14-support

React 0.14 support

igorbt committed:
Merge branch 'slider-fix' of https://github.com/igorbt/material-ui into slider-fix

Conflicts:
	src/slider.jsx

igorbt commented:
@shaurya947, I just rebased. Can you please also have a look and merge this PR? That's a pain to keep rebasing given the change rate in the library.

shaurya947 commented:
There you go, thanks @igorbt

igorbt commented:
Awsome! Thanks @shaurya947! I think at least 5 bugs shoul be closed now.
